### PR TITLE
 [Feature] Support LoveDA dataset

### PR DESCRIPTION
**Pull request by Junjue-Wang**
Old pr of LoveDA dataset is here: https://github.com/open-mmlab/mmsegmentation/pull/1006.

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Junjue-Wang** [comment]
@MengzhangL I have already created a new pr with a new branch.

**Comment by MengzhangLI** [comment]
Here is some results on validation and test set:
Val mIoU | Test mIoU | Model  | backbone  | Lr schd
-- | -- | -- | -- | --
50.28 | 47.51 | DeepLabV3+ | R-18-D8 | 80k
50.99 | 50.22 | DeepLabV3+ | R-50-D8 | 80k
51.47 | 49.60 | DeepLabV3+ | R-101-D8 | 80k
48.62 | 47.17 | PSPNet | R-18-D8 | 80k
50.46 | 50.32 | PSPNet | R-50-D8 | 80k
51.86 | 50.37 | PSPNet | R-101-D8 | 80k
49.3 | 47.57 | HRNet FCN | HRNetV2p-W18-Small | 80k
50.87 | 49.25 | HRNet FCN | HRNetV2p-W18 | 80k
51.04 | 50.38 | HRNet FCN | HRNetV2p-W48 | 80k

FYI, the results of original paper:
image


TO DO:

- [x] Getting results of models above on test set.

- [ ] Adding `format-only` in `./mmseg/datasets/loveda.py`, enabling users of MMSegmentation could directly generating standard prediction output of test set for evaluation on  [official codaLab site](https://competitions.codalab.org/competitions/35865#).

**Comment by Junjue-Wang** [comment]
> Here is some results on validation and test set:
> 
> Val mIoU	Test mIoU	Model	backbone	Lr schd
> 50.28	47.51	DeepLabV3+	R-18-D8	80k
> 50.99	50.22	DeepLabV3+	R-50-D8	80k
> 51.47	49.60	DeepLabV3+	R-101-D8	80k
> 48.62	47.17	PSPNet	R-18-D8	80k
> 50.46	50.32	PSPNet	R-50-D8	80k
> 51.86	50.37	PSPNet	R-101-D8	80k
> 49.3	47.57	HRNet FCN	HRNetV2p-W18-Small	80k
> 50.87	49.25	HRNet FCN	HRNetV2p-W18	80k
> 51.04	50.38	HRNet FCN	HRNetV2p-W48	80k
> FYI, the results of original paper: image
> 
> TO DO:
> 
> * [x]  Getting results of models above on test set.
> * [ ]  Adding `format-only` in `./mmseg/datasets/loveda.py`, enabling users of MMSegmentation could directly generating standard prediction output of test set for evaluation on  [official codaLab site](https://competitions.codalab.org/competitions/35865#).

Good job! This looks better than my test results, perhaps because of more data enhancement and iteration steps.

**Comment by MengzhangLI** [comment]
> > Here is some results on validation and test set:
> > Val mIoU	Test mIoU	Model	backbone	Lr schd
> > 50.28	47.51	DeepLabV3+	R-18-D8	80k
> > 50.99	50.22	DeepLabV3+	R-50-D8	80k
> > 51.47	49.60	DeepLabV3+	R-101-D8	80k
> > 48.62	47.17	PSPNet	R-18-D8	80k
> > 50.46	50.32	PSPNet	R-50-D8	80k
> > 51.86	50.37	PSPNet	R-101-D8	80k
> > 49.3	47.57	HRNet FCN	HRNetV2p-W18-Small	80k
> > 50.87	49.25	HRNet FCN	HRNetV2p-W18	80k
> > 51.04	50.38	HRNet FCN	HRNetV2p-W48	80k
> > FYI, the results of original paper: image
> > TO DO:
> > 
> > * [x]   Getting results of models above on test set.
> > * [ ]   Adding `format-only` in `./mmseg/datasets/loveda.py`, enabling users of MMSegmentation could directly generating standard prediction output of test set for evaluation on  [official codaLab site](https://competitions.codalab.org/competitions/35865#).
> 
> Good job! This looks better than my test results, perhaps because of more data enhancement and iteration steps.

I just only test results of HRNet, PSPNet and DeepLabV3Plus which are showed in your table. If your time available, you could train transformer models such as SegFormer, SETR and Swin Transformer which may get better results I guess.

**Comment by MengzhangLI** [comment]
Hi, Junjue.
I think I've done most part of this pr.

Could you please:

(1) Checking out my README.md and results. For example, your test results could be generated by `format-only` command and be like:

> python ./tools/test.py ./configs/pspnet/pspnet_r50-d8_512x512_80k_loveda.py ./work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth --format-only --eval-options "imgfile_prefix=./pspnet_test_results"

(2) Adding links on your official repo just like [SETR](https://github.com/fudan-zvg/SETR) and [SegFormer](https://github.com/NVlabs/SegFormer) so other people may use our codebase to set up their own benchmark, be like:


image

Best,

**Comment by MengzhangLI** [comment]
@Junjue-Wang I am not sure whether this version is up-to-merge. Could you please check whether it could be run from (1) LoveDA data preparation to (2) inference using our provided model and config and getting (3) test results in your website?

Thank you so much.

If it works well, you could also use this codebase to train better benchmark for LoveDA.

File: configs/hrnet/fcn_hr18s_512x512_80k_loveda.py [hunk 1]
```diff
@@ -1,7 +1,6 @@
 _base_ = './fcn_hr18_512x512_80k_loveda.py'
 model = dict(
-    # pretrained='open-mmlab://msra/hrnetv2_w18_small',
-    pretrained='./pretrained/hrnetv2_w18_small-b5a04e21.pth',
+    pretrained='open-mmlab://msra/hrnetv2_w18_small',
```
Comment (Junjun2016) [inline configs/hrnet/fcn_hr18s_512x512_80k_loveda.py:3]: Use init_cfg

File: configs/hrnet/fcn_hr48_512x512_80k_loveda.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 _base_ = './fcn_hr18_512x512_80k_loveda.py'
 model = dict(
-    # pretrained='open-mmlab://msra/hrnetv2_w48',
-    pretrained='./pretrained/hrnetv2_w48-d2186c55.pth',
+    pretrained='open-mmlab://msra/hrnetv2_w48',
```
Comment (Junjun2016) [inline configs/hrnet/fcn_hr48_512x512_80k_loveda.py:3]: Use init_cfg

File: configs/deeplabv3plus/deeplabv3plus_r101-d8_512x512_80k_loveda.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+_base_ = './deeplabv3plus_r50-d8_512x512_80k_loveda.py'
+model = dict(pretrained='open-mmlab://resnet101_v1c', backbone=dict(depth=101))
```
Comment (Junjun2016) [inline configs/deeplabv3plus/deeplabv3plus_r101-d8_512x512_80k_loveda.py:2]: Use init_cfg

File: configs/deeplabv3plus/deeplabv3plus_r18-d8_512x512_80k_loveda.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+_base_ = './deeplabv3plus_r50-d8_512x512_80k_loveda.py'
+model = dict(
+    pretrained='open-mmlab://resnet18_v1c',
```
Comment (Junjun2016) [inline configs/deeplabv3plus/deeplabv3plus_r18-d8_512x512_80k_loveda.py:3]: Use init_cfg

**Comment by Junjue-Wang** [comment]
> Hi, Junjue @Junjue-Wang I think I've done most part of this pr.
> 
> Could you please:
> 
> (1) Checking out my README.md and results. For example, your test results could be generated by `format-only` command and be like:
> 
> > python ./tools/test.py ./configs/pspnet/pspnet_r50-d8_512x512_80k_loveda.py ./work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth --format-only --eval-options "imgfile_prefix=./pspnet_test_results"
> 
> (2) Adding links on your official repo just like [SETR](https://github.com/fudan-zvg/SETR) and [SegFormer](https://github.com/NVlabs/SegFormer) so other people may use our codebase to set up their own benchmark, be like:
> 
> image
> 
> Best,

I have already checked the dataset LoveDA data preparation and fix some errors in [dataset_prepare.md](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs/dataset_prepare.md). Could you please send me the pre-trained weights './work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth' so that I can further check the inference process.

File: configs/pspnet/pspnet_r101-d8_512x512_80k_loveda.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+_base_ = './pspnet_r50-d8_512x512_80k_loveda.py'
+model = dict(pretrained='open-mmlab://resnet101_v1c', backbone=dict(depth=101))
```
Comment (Junjun2016) [inline configs/pspnet/pspnet_r101-d8_512x512_80k_loveda.py:2]: Use init_cfg

File: configs/pspnet/pspnet_r18-d8_512x512_80k_loveda.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+_base_ = './pspnet_r50-d8_512x512_80k_loveda.py'
+model = dict(
+    pretrained='open-mmlab://resnet18_v1c',
```
Comment (Junjun2016) [inline configs/pspnet/pspnet_r18-d8_512x512_80k_loveda.py:3]: Use init_cfg

File: docs_zh-CN/dataset_prepare.md [hunk 7]
```diff
@@ -195,3 +195,37 @@ python tools/convert_datasets/stare.py /path/to/stare-images.tar /path/to/labels
 ### Nighttime Driving
 
 因为我们只支持在此数据集上测试模型，所以您只需下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip)。
+
+### LoveDA
+
+下载[LoveDA数据集](https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing)。
+
+对于 LoveDA 数据集，请运行以下命令下载并组织数据集
```
Comment (Junjun2016) [inline docs_zh-CN/dataset_prepare.md:203]:
```suggestion
对于 LoveDA 数据集，请运行以下命令下载并重新组织数据集
```

**Comment by Junjun2016** [comment]
> > Hi, Junjue @Junjue-Wang I think I've done most part of this pr.
> > Could you please:
> > (1) Checking out my README.md and results. For example, your test results could be generated by `format-only` command and be like:
> > > python ./tools/test.py ./configs/pspnet/pspnet_r50-d8_512x512_80k_loveda.py ./work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth --format-only --eval-options "imgfile_prefix=./pspnet_test_results"
> > 
> > 
> > (2) Adding links on your official repo just like [SETR](https://github.com/fudan-zvg/SETR) and [SegFormer](https://github.com/NVlabs/SegFormer) so other people may use our codebase to set up their own benchmark, be like:
> > image
> > Best,
> 
> I have already checked the dataset LoveDA data preparation and fix some errors in [dataset_prepare.md](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs/dataset_prepare.md). Could you please send me the pre-trained weights './work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth' so that I can further check the inference process.

Can download from the links in README.

**Comment by Junjue-Wang** [comment]
> > > Hi, Junjue @Junjue-Wang I think I've done most part of this pr.
> > > Could you please:
> > > (1) Checking out my README.md and results. For example, your test results could be generated by `format-only` command and be like:
> > > > python ./tools/test.py ./configs/pspnet/pspnet_r50-d8_512x512_80k_loveda.py ./work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth --format-only --eval-options "imgfile_prefix=./pspnet_test_results"
> > > 
> > > 
> > > (2) Adding links on your official repo just like [SETR](https://github.com/fudan-zvg/SETR) and [SegFormer](https://github.com/NVlabs/SegFormer) so other people may use our codebase to set up their own benchmark, be like:
> > > image
> > > Best,
> > 
> > 
> > I have already checked the dataset LoveDA data preparation and fix some errors in [dataset_prepare.md](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs/dataset_prepare.md). Could you please send me the pre-trained weights './work_dirs/temp/pspnet_r50-d8_512x512_80k_loveda/iter_80000.pth' so that I can further check the inference process.
> 
> Can download from the links in README.

Thank you for your help. I have already checked the overall pipeline and successfully reproduced the PSPNet-R50 results on the website~ I have fixed some path errors. Thanks for providing the more powerful benchmark results @MengzhangLI.  I'll add the links to my repo.

**Comment by Junjue-Wang** [comment]
@MengzhangLI I have found that the title sizes of LoveDA seem inconsistent with the others in README.md.

**Comment by MengzhangLI** [comment]
> @MengzhangLI I have found that the title sizes of LoveDA seem inconsistent with the others in README.md.

(1) I checked that title size, it is caused by only Cityscapes and ADE20K using `###` but other dataset using `####`. So I will make another pr to fix this problem.

(2) Thanks for your recognition on our work. I just pushed again using `init_cfg` to replace `pretrained`. Could you also pulled it down to testify whether it could be trained correctly (not too much iteration times, only successful training process needed)?

Best,

**Comment by Junjue-Wang** [comment]
> > @MengzhangLI I have found that the title sizes of LoveDA seem inconsistent with the others in README.md.
> 
> (1) I checked that title size, it is caused by only Cityscapes and ADE20K using `###` but other dataset using `####`. So I will make another pr to fix this problem.
> 
> (2) Thanks for your recognition on our work. I just pushed again using `init_cfg` to replace `pretrained`. Could you also pulled it down to testify whether it could be trained correctly (not too much iteration times, only successful training process needed)?
> 
> Best,
I have trained the pspnet-r50  on one GPU successfully. And it seems good.
image

**Comment by MengzhangLI** [comment]
> > > @MengzhangLI I have found that the title sizes of LoveDA seem inconsistent with the others in README.md.
> > 
> > 
> > (1) I checked that title size, it is caused by only Cityscapes and ADE20K using `###` but other dataset using `####`. So I will make another pr to fix this problem.
> > (2) Thanks for your recognition on our work. I just pushed again using `init_cfg` to replace `pretrained`. Could you also pulled it down to testify whether it could be trained correctly (not too much iteration times, only successful training process needed)?
> > Best,
> > I have trained the pspnet-r50  on one GPU successfully. And it seems good.
> > image

Happy to see it training well, so my last thing is adding unit test for its coverage rate. 

image

**Comment by Junjue-Wang** [comment]
> > > > @MengzhangLI I have found that the title sizes of LoveDA seem inconsistent with the others in README.md.
> > > 
> > > 
> > > (1) I checked that title size, it is caused by only Cityscapes and ADE20K using `###` but other dataset using `####`. So I will make another pr to fix this problem.
> > > (2) Thanks for your recognition on our work. I just pushed again using `init_cfg` to replace `pretrained`. Could you also pulled it down to testify whether it could be trained correctly (not too much iteration times, only successful training process needed)?
> > > Best,
> > > I have trained the pspnet-r50  on one GPU successfully. And it seems good.
> > > image
> 
> Happy to see it training well, so my last thing is adding unit test for its coverage rate.
> 
> image

Good job! Best wishes~

**Comment by RockeyCoss** [comment]
Please merge the master branch into your branch, thank you.

File: configs/pspnet/README.md [hunk 8]
```diff
@@ -98,15 +98,29 @@ We support evaluation results on these two datasets using models above trained o
...
+| PSPNet | R-101-D8 | 512x512   |   80000 | 9.61       | 4.58          | 51.86 |         51.34 | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/pspnet/pspnet_r101-d8_512x512_80k_loveda.py) | [model](https://download.openmmlab.com/mmsegmentation/v0.5/pspnet/pspnet_r101-d8_512x512_80k_loveda/pspnet_r101-d8_512x512_80k_loveda_20211104_153212-1c06c6a8.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/pspnet/pspnet_r101-d8_512x512_80k_loveda/pspnet_r101-d8_512x512_80k_loveda_20211104_153212.log.json) |
 | PSPNet | R-50-D8  | 512x512   |   80000 | 9.6      | 20.5          | 38.80 |         39.19 | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/pspnet_r50-d8_512x512_4x4_80k_coco-stuff164k.py)   | [model](https://download.openmmlab.com/mmsegmentation/v0.5/pspnet/pspnet_r50-d8_512x512_4x4_80k_coco-stuff164k/pspnet_r50-d8_512x512_4x4_80k_coco-stuff164k_20210707_152034-0e41b2db.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/pspnet/pspnet_r50-d8_512x512_4x4_80k_coco-stuff164k/pspnet_r50-d8_512x512_4x4_80k_coco-stuff164k_20210707_152034.log.json)         |
```
Comment (Junjun2016) [inline configs/pspnet/README.md:124]: Duplicated coco-stuff 164k results.

File: docs/dataset_prepare.md [hunk 9]
```diff
@@ -253,3 +253,32 @@ Since we only support test models on this dataset, you may only download [the va
...
+
+The data could be downloaded [here](https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing).
+
+For LoveDA dataset, please run the following command to download and re-organize the dataset.
+
+```shell
+# download
+mkdir -p data/loveDA && cd data/loveDA
+wget https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing
+
```
Comment (Junjun2016) [inline docs/dataset_prepare.md:267]:
```suggestion
```

File: docs_zh-CN/dataset_prepare.md [hunk 10]
```diff
@@ -195,3 +195,33 @@ python tools/convert_datasets/stare.py /path/to/stare-images.tar /path/to/labels
...
 因为我们只支持在此数据集上测试模型，所以您只需下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip)。
+
+### LoveDA
+
+下载[LoveDA数据集](https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing)。
+
+对于 LoveDA 数据集，请运行以下命令下载并重新组织数据集
+
+```shell
+
```
Comment (Junjun2016) [inline docs_zh-CN/dataset_prepare.md:206]:
```suggestion
```

File: docs_zh-CN/dataset_prepare.md [hunk 11]
```diff
@@ -195,3 +195,33 @@ python tools/convert_datasets/stare.py /path/to/stare-images.tar /path/to/labels
...
+# download
+mkdir -p data/loveDA && cd data/loveDA
+wget https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing
+
```
Comment (Junjun2016) [inline docs_zh-CN/dataset_prepare.md:210]:
```suggestion
```

File: mmseg/datasets/loveda.py [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+            indices (list[int], optional): Indices of input results, if not
+                set, all the indices of the dataset will be used.
+                Default: None.
+
+        Returns:
+            list[str: str]: result txt files which contains corresponding
+            semantic segmentation images.
+        """
+        if indices is None:
+            indices = list(range(len(self)))
```
Comment (Junjun2016) [inline mmseg/datasets/loveda.py:51]: Not covered by unittests.

File: mmseg/datasets/loveda.py [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+        result_files = []
+        for result, idx in zip(results, indices):
+
+            filename = self.img_infos[idx]['filename']
+            basename = osp.splitext(osp.basename(filename))[0]
+
+            png_filename = osp.join(imgfile_prefix, f'{basename}.png')
+
+            # The  index range of official requirement is from 0 to 6.
+            output = Image.fromarray(result.astype(np.uint8))
```
Comment (Junjun2016) [inline mmseg/datasets/loveda.py:63]: Can result be `list | tuple | ndarray`? 
Refer to L37-38.

File: mmseg/datasets/loveda.py [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+        """
+        if indices is None:
+            indices = list(range(len(self)))
+
+        assert isinstance(results, list), 'results must be a list.'
+        assert isinstance(indices, list), 'indices must be a list.'
+
+        result_files = self.results2img(results, imgfile_prefix, indices)
+
+        return result_files
```
Comment (Junjun2016) [inline mmseg/datasets/loveda.py:94]: Should add doc for preparing the results and upload it to the official evaluation server (both command and description).

**Review by Junjun2016 — COMMENTED** [review]
Should add doc for preparing the results and upload it to the official evaluation server (both command and description).

**Comment by Junjun2016** [comment]
image

**Comment by MengzhangLI** [comment]
- [ ] LoveDA Unittest coverage rate.
- [ ] LoveDA transferring `.sh` shell file for submitting server.

**Comment by MengzhangLI** [comment]
Here are inference process and results of website:

image

image

**Review by Junjun2016 — APPROVED** [review]

**Comment by MengzhangLI** [comment]
Hi, @Junjue-Wang 

I have uploaded a convert .py file because `wget` can not download dataset directly from google drive.

Could you testify my convert file? Thanks!

Here is [file](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/tools/convert_datasets/loveda.py) and [doc](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs_zh-CN/dataset_prepare.md).

**Comment by Junjue-Wang** [comment]
> Hi, @Junjue-Wang
> 
> I have uploaded a convert .py file because `wget` can not download dataset directly from google drive.
> 
> Could you testify my convert file? Thanks!
> 
> Here is [file](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/tools/convert_datasets/loveda.py) and [doc](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs_zh-CN/dataset_prepare.md).

@MengzhangLI Thank you for your suggestion and modification, we have uploaded the dataset to https://zenodo.org/record/5706578#.YZumXmBByUk. Now we can directly use `wget` to download the LoveDA dataset. So I think you can easily change the URL in the README.md.

**Comment by MengzhangLI** [comment]
> > Hi, @Junjue-Wang
> > I have uploaded a convert .py file because `wget` can not download dataset directly from google drive.
> > Could you testify my convert file? Thanks!
> > Here is [file](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/tools/convert_datasets/loveda.py) and [doc](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs_zh-CN/dataset_prepare.md).
> 
> @MengzhangLI Thank you for your suggestion and modification, we have uploaded the dataset to https://zenodo.org/record/5706578#.YZumXmBByUk. Now we can directly use `wget` to download the LoveDA dataset. So I think you can easily change the URL in the README.md.

OK, got it. I will upload doc.

But it still needs dataset conversion. Can you have a try my .py file? Thanks!

**Comment by Junjue-Wang** [comment]
> > > Hi, @Junjue-Wang
> > > I have uploaded a convert .py file because `wget` can not download dataset directly from google drive.
> > > Could you testify my convert file? Thanks!
> > > Here is [file](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/tools/convert_datasets/loveda.py) and [doc](https://github.com/Junjue-Wang/mmsegmentation/blob/New_LoveDA/docs_zh-CN/dataset_prepare.md).
> > 
> > 
> > @MengzhangLI Thank you for your suggestion and modification, we have uploaded the dataset to https://zenodo.org/record/5706578#.YZumXmBByUk. Now we can directly use `wget` to download the LoveDA dataset. So I think you can easily change the URL in the README.md.
> 
> OK, got it. I will upload doc.
> 
> But it still needs dataset conversion. Can you have a try my .py file? Thanks!

Okay, I'll try the conversion py later :).

**Comment by MengzhangLI** [comment]
Hi, I have updated `wget` from zenodo in docs and also a convert dataset .py file.

File: docs_zh-CN/inference.md [hunk 15]
```diff
@@ -84,7 +84,7 @@ python tools/test.py ${配置文件} ${检查点文件} [--out ${结果文件}]
     ```
 
    您会在文件夹 `./pspnet_test_results` 里得到生成的 png 文件。
-   您也许可以运行 `zip -r results.zip pspnet_test_results/` 并提交 zip 文件给 [evaluation server](https://www.cityscapes-dataset.com/submit/)。
+   您也许可以运行 `zip -r results.zip pspnet_test_results/` 并提交 zip 文件给 [evaluation server](https://www.cityscapes-dataset.com/submit/).
```
Comment (xvjiarui) [inline docs_zh-CN/inference.md:87]:
```suggestion
   您也许可以运行 `zip -r results.zip pspnet_test_results/` 并提交 zip 文件给 [evaluation server](https://www.cityscapes-dataset.com/submit/)。
```

File: docs_zh-CN/dataset_prepare.md [hunk 16]
```diff
@@ -190,8 +198,33 @@ python tools/convert_datasets/stare.py /path/to/stare-images.tar /path/to/labels
 
 ### Dark Zurich
 
-因为我们只支持在此数据集上测试模型，所以您只需下载[验证集](https://data.vision.ee.ethz.ch/csakarid/shared/GCMA_UIoU/Dark_Zurich_val_anon.zip)。
+因为我们只支持在此数据集上测试模型，所以您只需下载[验证集](https://data.vision.ee.ethz.ch/csakarid/shared/GCMA_UIoU/Dark_Zurich_val_anon.zip) 。
 
 ### Nighttime Driving
 
-因为我们只支持在此数据集上测试模型，所以您只需下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip)。
+因为我们只支持在此数据集上测试模型，所以您只需下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip) 。
```
Comment (xvjiarui) [inline docs_zh-CN/dataset_prepare.md:205]:
```suggestion
因为我们只支持在此数据集上测试模型，所以您只需下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip)。
```

Comment (MengzhangLI) [reply docs_zh-CN/dataset_prepare.md:205]: Full width symbol would make hyperlink useless in markdown if it is close to hyperlink. So in this pr we unify all Chinese doc by adding a blank between them.

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**